modules/bigquery-dataset/variables.tf

```
@@ -164,6 +164,44 @@ variable "project_id" {
   type        = string
 }
 
+variable "routines" {
+  description = "Routine definitions."
+  type = map(object({
+    description          = optional(string)
+    routine_type         = string
+    language             = optional(string)
+    definition_body      = string
+    imported_libraries   = optional(list(string))
+    determinism_level    = optional(string)
+    data_governance_type = optional(string)
+    return_table_type    = optional(string)
+    arguments = optional(map(object({
+      argument_kind = optional(string)
+      mode          = optional(string)
+      data_type     = optional(string)
+    })), {})
+    spark_options = optional(list(object({
```

```suggestion
    spark_options = optional(object({
```
Does providing multiple `spark_options` really make sense?